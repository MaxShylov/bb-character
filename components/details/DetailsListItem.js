import { StyleSheet, Text, View } from 'react-native';

export const DetailsListItem = ({ data: { title, value } = {} }) => {
  return (
    <View style={styles.wrapper}>
      <Text style={styles.title}>{title}:</Text>
      <Text style={styles.value}>{value}</Text>
    </View>
  )
}

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginTop: 20,
  },
  title: {
    width: '30%',
    minWidth: 100,
    marginRight: 20,
    fontSize: 14,
  },
  value: {
    width: '60%',
    fontSize: 20,
    fontWeight: 'bold',
  },
});
