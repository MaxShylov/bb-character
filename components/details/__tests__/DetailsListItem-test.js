import renderer from 'react-test-renderer';
import { DetailsListItem } from '../DetailsListItem';

test('renders correctly', () => {
  const tree = renderer.create(<DetailsListItem />).toJSON();

  expect(tree).toMatchSnapshot();
});
