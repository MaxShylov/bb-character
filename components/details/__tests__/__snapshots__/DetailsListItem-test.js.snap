// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 14,
        "marginRight": 20,
        "minWidth": 100,
        "width": "30%",
      }
    }
  >
    :
  </Text>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "width": "60%",
      }
    }
  />
</View>
`;
